chain/chain-primitives/src/error.rs

```diff
@@ -304,6 +304,70 @@ impl Error {
             _ => false,
         }
     }
+
+    /// Some blockchain errors are reported in the prometheus metrics. In such cases a report might
+    /// contain a label that specifies the type of error that has occured. For example when the node
+    /// receives a block with missing chunks this would be reported as:
+    ///  `near_num_invalid_blocks{error="chunks_missing"}`.
+    /// This function returns the value of the error label for a specific instance of Error.
+    pub fn prometheus_label_value(&self) -> &'static str {
```

Comment:
This is not wrong, but you can make the compiler generate this for you, if you
`[derive(strum::Display)]` on the error, and then use `Display` to get the metric label.
The advantage of using the generated code is less maintenance and fewer typos.

Reply:
Thats pretty cool, thanks I will change it to `strum::Display`

Reply:
I'm a bit worried that the autogenerated display will also contain the debug information contained in the error. Please make sure that isn't the case.

Reply:
@wacban is right. `thiserror::Error` implements `Display`. Can't have two different implementations of `Display`.

Reply:
:+1:, trying to derive `strum::Display` fails:
```
error[E0119]: conflicting implementations of trait `std::fmt::Display` for type `error::Error`
  --> chain/chain-primitives/src/error.rs:61:35
   |
61 | #[derive(Debug, thiserror::Error, strum::Display)]
   |                 ----------------  ^^^^^^^^^^^^^^ conflicting implementation for `error::Error`
   |                 |
   |                 first implementation here
   |
   = note: this error originates in the derive macro `strum::Display` (in Nightly builds, run with -Z macro-backtrace for more info)
```